File: Data Structures & Algorithms/validate-parentheses/submission-7.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        closeToOpen = { ")" : "(", "]" : "[", "}" : "{" }
        for ss in s:
            if ss in closeToOpen:
                if  len(stack)==0 or  closeToOpen[ss] !=stack.pop():
                    return False
            else:
                stack.append(ss)
        return len(stack)==0

File: Data Structures & Algorithms/minimum-stack/submission-7.cpp
class MinStack {
    stack<int> st;
    stack<int>minSt;
public:
    MinStack() {
       
    }
    
    void push(int val) {
        if(!minSt.empty()){
            minSt.push(min(val,minSt.top()));
        }else{
            minSt.push(val);
        }
        st.push(val);
    }
    
    void pop() {
        st.pop();
        minSt.pop();
    }
    
    int top() {
        return st.top();
    }
    
    int getMin() {
        return minSt.top();
    }
};
